type script/unAuthorized ~@Error.ts

```diff
@@ -0,0 +1,28 @@
+import { Inject, Optional } from "@angular/core";
+import { LocalStorageService } from "@core/services/local-storage.service";
+import { ModalService } from "src/app/shared/services/modal.service";
+import { ModalConstants } from "src/app/utils/modals-constants";
+
+export class UnAuthorizedError {
+
+    constructor(@Optional() @Inject(ModalService) private _modalService: ModalService,
+    @Optional() @Inject(LocalStorageService) private _localStorageService: LocalStorageService) {
+        this.handleUnauthorizedError();
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider moving async call from constructor</b></div>
<div id="fix">

The `handleUnauthorizedError` method is being called in the constructor which could lead to side effects during object initialization. Consider moving this call to a separate initialization method or where the error actually occurs.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -8,9 +8,8 @@
      constructor(@Optional() @Inject(ModalService) private _modalService: ModalService,
          @Optional() @Inject(LocalStorageService) private _localStorageService: LocalStorageService) {
 -        this.handleUnauthorizedError();
      }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/24#issuecomment-2514594326>#d24210</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged